current/runtime/fees.md

```diff
@@ -146,15 +202,53 @@ validators to abuse these types of dispatches, they may not be included in block
 errors. This dispatch class exists to serve the assumption that it is better to allow an overweight
 block to be created than to not allow any block to be created at all.
 
+### Dynamic Weights
+
+In addition to purely fixed weights and constants it is possible to base the weight calculation
+on the input arguments of the dispatchable in question. It is important to understand that the
+weight should be trivially computable from the input arguments with some basic arithmetic:
+
+```rust
+#[weight = FunctionOf(
+  |args: (&Vec<User>,)| args.0.len().saturating_mul(10_000),
+  DispatchClass::Normal,
+  Pays::Yes,
+)]
+fn handle_users(origin, calls: Vec<User>) {
+  // Do something per user
+}
+```
+
+## Post Dispatch Weight Correction
+
+A dispatchable can decide that its actual weight is smaller than the ahead of dispatch calculated
+weight. Why this is useful is explained in the
+[weights chapter](../learn-substrate/weights#Post_Dispatch_Weight_Correction). In order to do so
+the dispatchable declares a different return type and then returns its actual weight:
+
+```rust
+#[weight = 10_000 + 500_000_000]
+fn expensive_or_cheap(input: u64) -> DispatchResultWithPostInfo {
+  let was_heavy = do_calculation(input);
+
+  if (was_heavy) {
+    // None means "no correction" from the weight annocation.
+    Ok(None.into())
+  } else {
+    // Return the actual weight consumed.
+    Ok(Some(10_000).into())
+  }
+}
+```
+
 ## Custom Fees
 
 You can also define custom fee systems through custom weight functions or inclusion fee functions.
 
 ### Custom Weights
 
-Implementing a custom weight calculation function can vary in complexity.
-
-A weight calculation function must provide two trait implementations:
+Instead of using the default weight annotations described above one can create a custom weight
+calculation type. This type must implement the follow traits:
 
 - [`WeighData<T>`]: To determine the weight of the dispatch.
 - [`ClassifyDispatch<T>`]: To determine the class of the dispatch.
```

Comment:
I think it also needs to implement PaysFees

Reply:
That is correct. I will add it.

Reply:
Fixed.